### ISSUE TITLE
Issue with measuring work with constrained systems

### ISSUE THREAD

**Issue by jchodera** [choderalab/openmmtools#158]
See https://github.com/choderalab/integrator-benchmark/issues/12 where we noticed the `LangevinSplittingIntegrator` can incorrectly report the shadow work or heat if the initial velocities have not been constrained to remove components along constrained bonds.

The new `LangevinIntegrator` suffers from this issue as well and will need to be fixed (though `NonequilibriumLangevinIntegrator` is fine.)

We have a few choices for fixing it:
* Option 1: For the first step (`step = 0`), apply position and velocity constraints, as in the [`NonequilibriumLangevinIntegrator`](https://github.com/choderalab/openmmtools/blob/master/openmmtools/integrators.py#L1452-L1462). This works fine as long as the user doesn't suddenly randomize the velocities or change the configuration, but can be dangerous if they do so without constraining the positions and velocities.
* Option 2: Apply position and velocity constraints at the beginning of each substep when requested to measure work or heat. This has the potential to significantly increase integrator overhead, though it's possible we can have OpenMM add some bookkeeping to keep track of whether anything has changed since the last time constraints were applied to make this essentially free. (That would be an OpenMM optimization feature request.)

Thoughts/feedback?

**Comment by jchodera** [comment]
After further consideration, `Context.setVelocitiesToTemperature()` actually *does* remove velocity components along constraints, so the only remaining is if one feeds non-constrained initial positions or velocities, the shadow work could wrong initially.

**Comment by maxentile** [comment]
I would say we should assume the initial positions and velocities are correctly constrained here

**Comment by jchodera** [comment]
I don't think we can assume that, since this is something that people will be using in general for their simulations. I think we at least need Option 1 (which we already do for `NonequilibriumLangevinIntegrator` anyway).
